public/scripts/request.js

```diff
@@ -0,0 +1,41 @@
+export const newGameRequest = (id, state, callback) => {
+  let payload = jsonPostPayload;
+  payload.body = state;
+  payload.body.gameType = id;
+  payload.body = JSON.stringify(payload.body);
+  makeRequest(endpoints.newGame, payload, callback);
+}
+
+export const makeMoveRequest = (idx, state, callback) => {  
+  let payload = jsonPostPayload;
+  payload.body = state;
+  payload.body.selectedIdx = idx;
+  payload.body = JSON.stringify(payload.body);
+
+  makeRequest(endpoints.move, payload, callback);
+}
+
+export const makeAiMoveRequest = (state, callback) => {
+  let payload = jsonPostPayload;
+  payload.body = JSON.stringify(state);
+  makeRequest(endpoints.move, payload, callback);
+}
+
+const endpoints = {
+  move: "/api/move",
+  newGame: "/api/new-game",
+}
+
+const makeRequest = (url, payload, callback) => {
+  fetch(url, payload)
+    .then(response => response.json()) 
+    .then(data => callback(data));
```

Comment:
How would you handle errors in the event that the response is not a success?

Reply:
I've altered the `#makeRequest` method to pass the `response` to the callback. The callback then determines what to do depending on the value of `response.ok()`. The method invoked by the callback can then asynchronously call `.json()` and make use of the JSON body.